src/core/Extract.ts

```diff
@@ -74,30 +74,27 @@ export async function extractHardStrings(document: TextDocument, extracts: Extra
 
   extracts.sort((a, b) => b.range.start.compareTo(a.range.start))
 
-  await Promise.all(
-    [
-      // replace
-      editor.edit((editBuilder) => {
-        for (const extract of extracts) {
-          editBuilder.replace(
-            extract.range,
-            extract.replaceTo,
-          )
-        }
-      }),
-      // save keys
-      CurrentFile.loader.write(
-        extracts
-          .filter(i => i.keypath != null && i.message != null)
-          .map(e => ({
-            textFromPath: filepath,
-            filepath: undefined,
-            keypath: e.keypath!,
-            value: e.message!,
-            locale: e.locale || sourceLanguage,
-          })),
-      ),
-    ],
+  // replace
+  await editor.edit((editBuilder) => {
+    for (const extract of extracts) {
+      editBuilder.replace(
+        extract.range,
+        extract.replaceTo,
+      )
+    }
+  })
+
+  // save keys
+  await CurrentFile.loader.write(
+    extracts
+      .filter(i => i.keypath != null && i.message != null)
+      .map(e => ({
+        textFromPath: filepath,
+        filepath: undefined,
+        keypath: e.keypath!,
+        value: e.message!,
+        locale: e.locale || sourceLanguage,
+      })),
```

Comment:
Needed this change because when extracting strings from SFC files, the file that is being edited is the target file where strings are extracted to. Needed to wait for a key replacement to finish to have the file in the correct state.